saviSimulation/savi/commandStation/CommandStationCore.java

```diff
@@ -49,38 +57,23 @@ public synchronized void sendMessage(String destination, String messageType, Str
 		// Build the message
 		String message = "<" + mid + "," + this.id + "," + messageType + "," + destination + "," + parameter + ">";
 		
-		// *** SEND OVER SOCKET HERE *** //
-		
-		// Send the message
-		this.sendMessagePassthrough(message);
-	}
-	
-	
-	/**
-	 * Sends the message to the agents
-	 * 
-	 * *** Assuming that the messages are valid, socket receiver should call this method ***
-	 * 
-	 * @param	The message to be sent - MUST BE A VALID MESSAGE, NO ERROR CHECKING IN THIS METHOD
-	 */
-	public synchronized void sendMessagePassthrough(String message) {
 		// Send the message
-		this.connector.messageOut(message);
+		socketConnector.messageOut(message);
 	}
-	
+		
 	
 	/**
 	 * Method to receive messages from the simulator
 	 * 
 	 * @param msg an incoming message.
 	 */
-	public void receiveMessage(String msg) {
-		
-		// *** SEND OVER SOCKET HERE *** //
-		
+	@Override
+	public void messageIn(String msg) {
+
+
 		// Parse the message and get a telemetryItem
 		TelemetryItem item = TelemetryItem.generateTelemetryItem(msg);
-		
+
 		// Parse the message
 		String parsedMessage;
 		if (item instanceof ThreatTelemetry) {
```

Comment:
These casts are unnecessary if all we need to do is call toString(). Is there another reason why we explicitly cast the telemetry item to ThreatTelemetry/PositionTelemetry/etc. ?

Reply:
There is a reason, I'll answer offline.